src/skadi/sample.cc

```diff
@@ -35,10 +34,43 @@ constexpr int16_t NO_DATA_HIGH = 16384;
 constexpr int16_t NO_DATA_LOW = -16384;
 constexpr size_t TILE_COUNT = 180 * 360;
 constexpr int8_t UNPACKED_TILES_COUNT = 50;
+const constexpr char* kDataPathPattern = "{DataPath}";
 
-// macro is faster than inline function for this..
+// macro is faster than inline function for this...
 #define out_of_range(v) v > NO_DATA_HIGH || v < NO_DATA_LOW
 
+template <typename T>
+void parallel_call(const std::function<double(const T&)>& func,
+                   std::vector<T> st,
+                   std::vector<double>& result) {
+  if (!func || st.empty())
+    return;
+
+  std::uint32_t size = std::max(1U, std::thread::hardware_concurrency());
```

Comment:
its fine to use max parallelism as a default but this needs to be from config otherwise.